util/buildkitskipper/hasher/hasher.go

```diff
@@ -30,24 +28,20 @@ func (h *Hasher) GetHash() []byte {
 	return h.h.Sum(nil)
 }
 
-func (h *Hasher) HashCommand(cmd spec.Command) {
-	dt, err := json.Marshal(cmd)
-	if err != nil {
-		panic(fmt.Sprintf("failed to hash command: %s", err)) // shouldn't happen
-	}
-	h.HashBytes(dt)
+func (h *Hasher) HashInt(i int) {
+	h.HashString(fmt.Sprintf("int:%d", i))
```

Comment:
something seems odd with this, as we're going to end up hashing `str:int:123`.

I think it makes more sense to call `h.h.Write([]byte(fmt.Sprintf("int:%d", i)))` directly, so we don't end up with cases where we could easily end up with hashing collisions.

e.g. currently we have a collision between `HashInt(123)` and `HashString("int:123")`.

It would be a good idea to look over the API of this package to see if there's an potentials for collisions.

Reply:
Updated.